### PR TITLE
fix(api):  bug bash preview issues resolved

### PR DESCRIPTION
**Pull request by tatarco**
### What changed? Why was the change needed?
During the bug bash the preview endpoint appeared to be very fragile, attempt to make it more robust was difficult due to the structure of the use cases, multiple linear bugs resolved + a massive cleanup and alignment to consistent company standards 
what was done? 

1. Pipe {{ttt | upper}} not converting to payload correctly.
2.. {{}} failing preview 
3.. Variable validation agains Available Variable schema (validate {{name}} illegal.
4.. Control values sanitation on preview ( illegal placeholders will be removed for preview but persisted along with errors in the DB to keep the UX consistent 
5.. computed Payload schema is now  persisted (as string ) on upsert and served from the DB on get test data 
 

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: .source
Comment (djabarovgeorge) [inline .source:1]: If we don't have changes in ee in this PR we should not have change here.

File: .cspell.json [hunk 1]
```diff
@@ -789,6 +789,7 @@
     "packages/shared/src/types/timezones/timezones.types.ts",
     "*.riv",
     "websockets",
+    "apps/api/src/app/workflows-v2/usecases/validate-content/validate-placeholders/validate-placeholder.usecase.ts",
```
Comment (djabarovgeorge) [inline .cspell.json:792]: can you add `/* cspell:disable */` on this file, i would not put here files that can be moved in the source code in the future.

File: apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema.usecase.ts [hunk 2]
```diff
@@ -9,16 +9,19 @@ export class ExpandEmailEditorSchemaUsecase {
   constructor(private hydrateEmailSchemaUseCase: HydrateEmailSchemaUseCase) {}
 
   execute(command: ExpandEmailEditorSchemaCommand): TipTapNode {
+    console.log('PRE:hydratedEmailSchema\n\n', JSON.stringify(command.emailEditorJsonAsString));
```
Comment (djabarovgeorge) [inline apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema.usecase.ts]: Did we decide to start console log actions?

```suggestion
```

File: apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema.usecase.ts [hunk 3]
```diff
@@ -9,16 +9,19 @@ export class ExpandEmailEditorSchemaUsecase {
...
 
     return emailSchemaHydrated;
   }
   private hydrate(command: ExpandEmailEditorSchemaCommand) {
     const { hydratedEmailSchema } = this.hydrateEmailSchemaUseCase.execute({
-      emailEditor: command.body,
+      emailEditor: command.emailEditorJsonAsString,
       fullPayloadForRender: command.fullPayloadForRender,
     });
+    console.log('hydratedEmailSchema\n\n', JSON.stringify(hydratedEmailSchema));
```
Comment (djabarovgeorge) [inline apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema.usecase.ts]: Same here.

```suggestion
```

File: apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema-command.ts [hunk 4]
```diff
@@ -4,6 +4,6 @@ import { BaseCommand } from '@novu/application-generic';
 import { FullPayloadForRender } from './render-command';
 
 export class ExpandEmailEditorSchemaCommand extends BaseCommand {
-  body: string;
+  emailEditorJsonAsString: string;
```
Comment (djabarovgeorge) [inline apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema-command.ts]: i think the type is enough to tell us it is a string, no? 

```suggestion
  emailEditorJson: string;
```

File: apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema.usecase.ts (hunk 2 above)
Comment (tatarco) [reply apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema.usecase.ts]: no no, left from the debugging

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**